### PR TITLE
Add fixes from travis (Closes #1312)

### LINKED ISSUE #1312: Possibly add Travis typos
Some may be Travis specific, but a lot are generic:
https://github.com/travis-ci/travis-yml/blob/master/var/keys.dict

Any ideas on how copyright applies to this? They're MIT licenced.
https://github.com/travis-ci/travis-yml/blob/master/LICENSE

### PR DESCRIPTION
**Pull request by skangas**
In #1312, it was discussed that it might be worth adding the fixes from Travis CI. This PR attempts to do that, while removing anything that I considered too Travis specific.

Note that many of these are variations on a small number of words (e.g. "language", "script"), yet seem useful enough to me as these words should be extremely commonly used in the context of software.

I've also added in a great many plural forms, variations, etc.

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -7337,6 +7397,8 @@ comptible->compatible
 comptue->compute
 compuatation->computation
 compuation->computation
+compuler->compiler
+compulers->compilers
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:7401]: `computer`, `computers`?

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -9186,6 +9252,8 @@ crewsant->croissant
 cricital->critical
 cricitally->critically
 cricitals->criticals
+cript->script
+cripts->scripts
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:9256]: `crypt`, `crypts`?

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -11519,6 +11597,7 @@ direcctries->directories
 direcdories->directories
 direcdory->directory
 direcdorys->directories
+direcetories->directories
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:11600]: `direcetory`?

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -12858,6 +12939,7 @@ efford->effort, afford,
 effords->efforts, affords,
 effulence->effluence
 eforceable->enforceable
+efore->before
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:12942]: `afore`?

I would like to be able to add some fixes in a dictionary of rare fixes...

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -30242,19 +30468,26 @@ screenchot->screenshot
...
 screenwrighter->screenwriter
 screnn->screen
+scriipt->script
+scriipted->scripted
+scriipting->scripting
 scriopted->scripted
 scriopting->scripting
 scriopts->scripts
 scriopttype->scripttype
+scripe->script
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:30478]: `scribe`, `scrape`, `scrip`, `stripe`

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -30242,19 +30468,26 @@ screenchot->screenshot
...
 scriping->scripting
 scripst->scripts
+scripte->script, scripted,
 scriptype->scripttype
+scrit->script
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:30483]: `scrip`

File: codespell_lib/data/dictionary.txt [hunk 7]
```diff
@@ -31469,6 +31730,12 @@ socail->social
 socalism->socialism
 socekts->sockets
 socities->societies
+socript->script
+socripted->scripted
+socripting->scripting
+socripts->scripts
+sodo->sudo
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:31737]: `soda`, `sod`, `sods`, `dodo`, `solo`

File: codespell_lib/data/dictionary.txt [hunk 8]
```diff
@@ -31469,6 +31730,12 @@ socail->social
...
+sodu->sudo
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:31738]: `soda`, `sod`, `sods`

File: codespell_lib/data/dictionary.txt [hunk 9]
```diff
@@ -31600,6 +31867,7 @@ soudns->sounds
 sould'nt->shouldn't
 sould->could, should, sold,
 souldn't->shouldn't
+soultion->solution
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:31870]: `soultions->solutions` (plural)

File: codespell_lib/data/dictionary.txt [hunk 10]
```diff
@@ -32994,10 +33270,14 @@ sucidial->suicidal
 sucome->succumb
 sucsede->succeed
 sucsess->success
+sude->sudo
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:33273]: `side`, `sure`, `dude`, `suede`, ...

File: codespell_lib/data/dictionary.txt [hunk 11]
```diff
@@ -35223,6 +35505,7 @@ udated->updated, dated,
 udater->updater, dater,
 udating->updating, dating,
 udno->undo, uno,
+udo->sudo
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:35508]: `undo`, `judo`, `udon`, `UFO`, `ado`, ...

File: codespell_lib/data/dictionary.txt [hunk 12]
```diff
@@ -37597,6 +37884,7 @@ xepects->expects
 xgetttext->xgettext
 xinitiazlize->xinitialize
 xmdoel->xmodel
+xode->xcode
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:37887]: plain `code`?

File: codespell_lib/data/dictionary.txt [hunk 13]
```diff
@@ -1184,6 +1191,7 @@ agrieved->aggrieved
 agrresive->aggressive
 agrument->argument
 agruments->arguments
+ags->tags
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:1194]: Added.

File: codespell_lib/data/dictionary.txt [hunk 14]
```diff
@@ -2943,6 +2953,8 @@ ascconciated->associated
 asceding->ascending
 ascpect->aspect
 ascpects->aspects
+ascript->script
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:2956]: Added.

File: codespell_lib/data/dictionary.txt [hunk 15]
```diff
@@ -4822,19 +4844,32 @@ boyant->buoyant
 boycot->boycott
 bracese->braces
 brach->branch
+brached->branched
+braches->branches
+braching->branching
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:4849]: Added.

File: codespell_lib/data/dictionary.txt [hunk 16]
```diff
@@ -5618,6 +5662,8 @@ ccertificates->certificates
 ccertification->certification
 ccessible->accessible
 cche->cache
+ccompiler->compiler
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:5665]: Added.

File: codespell_lib/data/dictionary.txt [hunk 17]
```diff
@@ -6145,6 +6191,7 @@ cilindrical->cylindrical
 cilyndre->cylinder
 cilyndres->cylinders
 cilyndrs->cylinders
+cimpiler->compiler
```
[the remark this answers is not in the document]
Comment (skangas) [reply codespell_lib/data/dictionary.txt:6194]: Added.

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:7401]: Added.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:9256]: Added.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:11600]: Added.

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:12942]: > `afore`?

Added.
 
> I would like to be able to add some fixes in a dictionary of rare fixes...

Yup, that would be useful.

File: codespell_lib/data/dictionary.txt (hunk 5 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:30478]: Added.

File: codespell_lib/data/dictionary.txt (hunk 6 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:30483]: Added.

File: codespell_lib/data/dictionary.txt (hunk 7 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:31737]: Good one. Added.

File: codespell_lib/data/dictionary.txt (hunk 8 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:31738]: Added.

File: codespell_lib/data/dictionary.txt (hunk 9 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:31870]: Added.

File: codespell_lib/data/dictionary.txt (hunk 10 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:33273]: Added. I also added `sued`.

File: codespell_lib/data/dictionary.txt (hunk 11 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:35508]: Added.

File: codespell_lib/data/dictionary.txt (hunk 12 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt:37887]: Yup. Added.

**Comment by skangas** [comment]
Thanks yet again for your patient review @DimitriPapadopoulos. I've added all your suggestions.

**Comment by DimitriPapadopoulos** [comment]
From https://github.com/codespell-project/codespell/issues/1312#issuecomment-545959606:
> My understanding is that you can use MIT code in GPL projects (MIT is compatible with use in GPL) but not the other way around. So as long as we add a note somewhere that the entries were adapted it seems fine

@larsoner Where could we add that we borrow fixes from Travis?

**Comment by DimitriPapadopoulos** [comment]
Perhaps acknowledge Travis and other dictionaries that have been used here, together with applicable license?
https://github.com/codespell-project/codespell#license

**Comment by skangas** [comment]
> From [#1312 (comment)](https://github.com/codespell-project/codespell/issues/1312#issuecomment-545959606):
> 
> > My understanding is that you can use MIT code in GPL projects (MIT is compatible with use in GPL) but not the other way around. So as long as we add a note somewhere that the entries were adapted it seems fine
> 
> @larsoner Where could we add that we borrow fixes from Travis?

You are of course correct that MIT is compatible with the GPL. The dictionary, however, is licensed under CC-BY-SA, so the GPL doesn't really enter into it. It must be considered separately.

The MIT license is quite permissive and allows you to redistribute it under whichever license you want - including a proprietary one! (That's why I prefer copyleft licenses like GPL myself.) So there is no problem whatsoever with us distributing this stuff under the CC-BY-SA license.

However, there is a demand that you do this, according to [the license](https://github.com/travis-ci/travis-yml/blob/master/LICENSE):

```
The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.
```

So to satisfy the license it's enough to include some kind of copyright blurb somewhere, for example in the README. Or we could perhaps just include it in a separate file in the dictionary folder. I'll think about it and come up with a proposal, unless anyone has any better ideas.

**Comment by DimitriPapadopoulos** [comment]
@skangas Can you rebase? I suggest we merge this as is, and address licences in #2682.

**Comment by skangas** [comment]
> @skangas Can you rebase? I suggest we merge this as is, and address licences in #2682.

Agreed. Now rebased.

**Review by DimitriPapadopoulos — APPROVED** [review]

File: codespell_lib/data/dictionary.txt [hunk 18]
```diff
@@ -35021,10 +35300,14 @@ sucidial->suicidal
 sucome->succumb
 sucsede->succeed
 sucsess->success
+sude->sudo, side, sure, dude, suede, sued,
 sudent->student
 sudents->students
+sudeo->sudo
+sudio->sudo
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:35307]: Perhaps `audio` too?

Comment (skangas) [reply codespell_lib/data/dictionary.txt]: Added.

File: codespell_lib/data/dictionary.txt [hunk 19]
```diff
@@ -36398,6 +36681,8 @@ tiemstamp->timestamp
 tiemstamped->timestamped
 tiemstamps->timestamps
 tieth->tithe
+tification->notification
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:36684]: This is strange, could also be `ratification`, `certification`, `fortification`, `justification`, `rectification`, `mortification`. Why this specific fix?

Ah wait, after looking it up in GitHub, it looks like some programmers might write “notification” as “no-tification”. For example:
https://github.com/mavenjan/BaseApp-master/blob/a5131f991afabc0c3a9058b258568245576179a6/mjlibrary/src/main/java/com/cauc/mavenj/utils/NotificationUtil.java#L22

What do you think?

Comment (skangas) [reply codespell_lib/data/dictionary.txt:36684]: > This is strange, could also be `identification`, `ratification`, `certification`, `fortification`, `justification`, `rectification`, `mortification`. Why this specific fix?

Hmm, you're right. 

> What do you think?

How about just adding in all the above alternatives? "tification" can never be right, in any case.

Comment (DimitriPapadopoulos) [reply codespell_lib/data/dictionary.txt:36684]: I don't know. This looks very specific. `tification` might appear when the complete word is split at the end of the line:
```
... no-
tification ...
```

Comment (skangas) [reply codespell_lib/data/dictionary.txt:36684]: There are a 20k hits here: https://github.com/search?q=tification&type=code

> `tification` might appear when the complete word is split at the end of the line:

Or in [quantification](https://github.com/sungunrun/index0926/blob/ad29eacd0a47fe312fecb055e12088dddde952ad/static/page_contents/test_196.html#L43), [justification](https://github.com/Deuy76/R_Textming_Tini/blob/c79aabf7dad6329d829c996792873ea181fa60c2/data/20news-bydate/20news-bydate-train/comp.os.ms-windows.misc/9996#L69), etc.

I'm honestly not sure. The options I see are either adding all variants, or deleting the entry.

Comment (larsoner) [reply codespell_lib/data/dictionary.txt:36684]: To keep it simple let's delete it. It's probably not sustainable to add truncated/split words

Comment (larsoner) [reply codespell_lib/data/dictionary.txt:36684]:
```suggestion
```

File: codespell_lib/data/dictionary.txt [hunk 20]
```diff
@@ -36398,6 +36681,8 @@ tiemstamp->timestamp
...
+tifications->notifications
```
Comment (larsoner) [inline codespell_lib/data/dictionary.txt:36685]:
```suggestion
```

File: codespell_lib/data/dictionary.txt (hunk 19 above)
Comment (skangas) [reply codespell_lib/data/dictionary.txt]: > To keep it simple let's delete it. It's probably not sustainable to add truncated/split words

Yup, makes sense.

**Pull request merged**